File: snippets-growing-app/src/main/kotlin/step2a/WhyTestsAreUseful.kt
package org.example.step2a

fun usageExample() {
    val workersProfit = listOf(100f, 200f, 80f, 10f)

    println(generateReportV1(workersProfit, workersProfit.sum()))
    println(generateReportV2(workersProfit, workersProfit.sum()))
}

fun generateReportV1(workersProfit: List<Float>, totalBusinessProfit: Float): List<Int> {
    val workersProfitPercentage =
        workersProfit.map { workerProfit -> (((workerProfit / workersProfit.size) % totalBusinessProfit)).toInt() }

    return workersProfitPercentage
}

fun generateReportV2(workersProfit: List<Float>, totalBusinessProfit: Float): List<Int> {
    val workersProfitPercentage =
        workersProfit.map { workerProfit -> (((workerProfit / workersProfit.size) % totalBusinessProfit)).toInt() }

    return workersProfitPercentage.map { workerProfitPercentage ->
        if (workerProfitPercentage <= 20) 0 else workerProfitPercentage
    }
}

File: snippets-growing-app-js/1-teste-com-testes/whyTestsAreUseful.ts
// input: [100, 200, 80, 10] -> result: [ 25, 50, 20, 2 ]
function generateReportV1(workersProfit: number[], totalBusinessProfit: number): number[] {
  const workersProfitPercentage: number[] = workersProfit.map(workerProfit =>
      Math.floor((workerProfit / workersProfit.length) % totalBusinessProfit)
  );

  return workersProfitPercentage;
}

// input: [100, 200, 80, 10] -> result: [ 25, 50, 0, 0 ]
function generateReportV2(workersProfit: number[], totalBusinessProfit: number): number[] {
  const workersProfitPercentage: number[] = workersProfit.map(workerProfit =>
      Math.floor((workerProfit / workersProfit.length) % totalBusinessProfit)
  );

  return workersProfitPercentage.map(workerProfitPercentage =>
      workerProfitPercentage <= 20 ? 0 : workerProfitPercentage
  );
}
